api/config/v1alpha1/envoyproxy_types.go

```diff
@@ -105,6 +109,85 @@ type ProxyLogging struct {
+	Format ProxyAccessLoggingFormat `json:"format"`
+	// Sink defines the sink of access logging.
+	Sink ProxyAccessLoggingSink `json:"sink"`
+}
+
+type ProxyAccessLoggingFormatType string
+
+const (
+	// ProxyAccessLoggingFormatTypeText defines the text access logging format.
+	ProxyAccessLoggingFormatTypeText ProxyAccessLoggingFormatType = "text"
+	// ProxyAccessLoggingFormatTypeJSON defines the JSON access logging format.
+	ProxyAccessLoggingFormatTypeJSON ProxyAccessLoggingFormatType = "json"
+	// TODO: support format type "mix" in the future.
+)
+
+type ProxyAccessLoggingFormat struct {
+	Type ProxyAccessLoggingFormatType `json:"type,omitempty"`
+	// Text defines the text access logging format, following Envoy access logging formatting,
+	// empty value results in proxy's default access log format.
+	// It's required when the format type is "text".
+	// Envoy [command operators](https://www.envoyproxy.io/docs/envoy/latest/configuration/observability/access_log/usage#command-operators) may be used in the format.
+	// The [format string documentation](https://www.envoyproxy.io/docs/envoy/latest/configuration/observability/access_log/usage#config-access-log-format-strings) provides more information.
+	// +optional
+	Text *string `json:"text,omitempty"`
+	// Fields is additional attributes that describe the specific event occurrence.
+	// Structured format for the envoy access logs. Envoy [command operators](https://www.envoyproxy.io/docs/envoy/latest/configuration/observability/access_log/usage#command-operators)
+	// can be used as values for fields within the Struct.
+	// It's required when the format type is "json".
+	// +optional
+	Fields *map[string]string `json:"fields,omitempty"`
+}
+
+type ProxyAccessLoggingSinkType string
+
+const (
+	// ProxyAccessLoggingSinkTypeFile defines the file access logging sink.
+	ProxyAccessLoggingSinkTypeFile ProxyAccessLoggingSinkType = "file"
+	// ProxyAccessLoggingSinkTypeOpenTelemetry defines the OpenTelemetry access logging sink.
+	ProxyAccessLoggingSinkTypeOpenTelemetry ProxyAccessLoggingSinkType = "opentelemetry"
+)
+
+type ProxyAccessLoggingSink struct {
+	// Type defines the type of access logging sink.
+	// +kubebuilder:validation:Enum=file;opentelemetry
+	Type ProxyAccessLoggingSinkType `json:"type,omitempty"`
+	// File defines the file access logging sink.
+	// +optional
+	File *FileEnvoyProxyAccessLogging `json:"file,omitempty"`
+	// OpenTelemetry defines the OpenTelemetry access logging sink.
+	// +optional
+	OpenTelemetry *OpenTelemetryEnvoyProxyAccessLogging `json:"opentelemetry,omitempty"`
+}
+
+type FileEnvoyProxyAccessLogging struct {
+	// Path defines the file path used to expose envoy access log(e.g. /dev/stdout).
+	// Empty value disables access logging.
+	Path string `json:"path,omitempty"`
+}
+
+// TODO: consider reuse ExtensionService?
+type OpenTelemetryEnvoyProxyAccessLogging struct {
+	// Host define the extension service hostname.
+	Host string `json:"host"`
```

Comment:
should this be URL instead ? which bakes in protocol, host and port in a single line ?

Reply:
it's grpc, use same filelds like `ExtensionService`, leave a TODO in L173.